src/Dialect/ONNX/Transforms/Recompose.cpp

```diff
@@ -602,6 +602,57 @@ struct RecomposeQLinearMatMulFromQuantizeLinearPattern
   }
 };
 
+/// Merges nested ONNXConcatOps
+struct RecomposeConcatPattern : public OpRewritePattern<ONNXConcatOp> {
```

Comment:
Could you move this pattern into Canonicalize.cpp? I would helpful to call this everytime we have concat. Putting it inside Recompose.cpp may only trigger at the beginning of the compilation.